packages/dds/matrix/src/matrix.ts

```diff
@@ -543,33 +644,82 @@ export class SharedMatrix<T = any>
 				);
 				break;
 			default: {
+				if (content.type === MatrixOp.changeSetCellPolicy) {
+					this.submitLocalMessage({
+						type: MatrixOp.changeSetCellPolicy,
+					});
+					break;
+				}
 				assert(
 					content.type === MatrixOp.set,
 					0x020 /* "Unknown SharedMatrix 'op' type." */,
 				);
 
 				const setOp = content as ISetOp<T>;
-				const { rowHandle, colHandle, localSeq, rowsRefSeq, colsRefSeq } =
-					localOpMetadata as ISetOpMetadata;
-
-				// If there are more pending local writes to the same row/col handle, it is important
-				// to skip resubmitting this op since it is possible the row/col handle has been recycled
-				// and now refers to a different position than when this op was originally submitted.
-				if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
+				const {
+					rowHandle,
+					colHandle,
+					localSeq,
+					rowsRefSeq,
+					colsRefSeq,
+					referenceSeqNumber,
+				} = localOpMetadata as ISetOpMetadata;
+
+				// Policy is not FWW. So behave accordingly.
+				if (this.setCellPolicySwitchOpSeqNumber > -1) {
 					const row = this.rebasePosition(this.rows, setOp.row, rowsRefSeq, localSeq);
 					const col = this.rebasePosition(this.cols, setOp.col, colsRefSeq, localSeq);
-
 					if (row !== undefined && col !== undefined && row >= 0 && col >= 0) {
-						this.sendSetCellOp(
-							row,
-							col,
-							setOp.value,
+						const lastCellModificationDetails = this.cellLastWriteTracker.getCell(
 							rowHandle,
 							colHandle,
-							localSeq,
-							rowsRefSeq,
-							colsRefSeq,
 						);
+						// If we generated this op, after seeing the last set op, or it is the first
+						// set op for the cell, then regenerate the op, otherwise raise conflict.
+						if (
+							lastCellModificationDetails === undefined ||
+							referenceSeqNumber >= lastCellModificationDetails.seqNum
+						) {
+							this.sendSetCellOp(
+								row,
+								col,
+								setOp.value,
+								rowHandle,
+								colHandle,
+								localSeq,
+								rowsRefSeq,
+								colsRefSeq,
+							);
+						} else {
+							this.emit(
+								"conflict",
+								row,
+								col,
+								this.cells.getCell(rowHandle, colHandle), // Current value
+								parseHandles(setOp.value, this.serializer), // Ignored value
+								this,
+							);
+						}
+					}
+				} else {
+					// If there are more pending local writes to the same row/col handle, it is important
+					// to skip resubmitting this op since it is possible the row/col handle has been recycled
+					// and now refers to a different position than when this op was originally submitted.
+					if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
```

Comment:
Can you please follow up with @DLehenbauer in that? While skipping ops in general seems wrong (even with LWW, as we promise we guarantee order of ops across container/different DDSs), it is unlikely to result in bugs. But we can't do it with FWW, so from that angle your code above (first IF branch looks correct). But if comment here is correct, then your code above looks wrong, as it could deal with wrong handles.
I do not fully understand the issue with handle reuse, Daniel is the rigth person to run it by.

Reply:
@DLehenbauer Should I call this.rows.getAllocatedHandle(adjusted row)/col to get correct handle in above If condition if that looks incorrect according to vlad's comment?

Reply:
Removed isLatestPendingWrite from resubmit flow after talking to Daniel as rebasePosition handles that and would return undefined if the handle is not valid. So, both branches are correct now and also refactored the code and simplified it.